Data/Functor/Foldable.hs

```diff
@@ -651,22 +658,22 @@ distGApoT g k = fmap ExceptT . k . fmap (distGApo g) . runExceptT
 histo :: Recursive t => (Base t (Cofree (Base t) a) -> a) -> t -> a
 histo = gcata distHisto
 
-ghisto :: (Recursive t, Functor h) => (forall b. Base t (h b) -> h (Base t b)) -> (Base t (Cofree h a) -> a) -> t -> a
+ghisto :: (Recursive t, Comonad w) => (forall b. Base t (w b) -> w (Base t b)) -> (Base t (CofreeT (Base t) w a) -> a) -> t -> a
 ghisto g = gcata (distGHisto g)
 
 distHisto :: Functor f => f (Cofree f a) -> Cofree f (f a)
-distHisto = distGHisto id
+distHisto fc = fmap extract fc :< fmap (distHisto . Cofree.unwrap) fc
```

Comment:
`distGHisto distCata` would be a valid implementation if `Cofree f` was defined as `CofreeT f Identity`